source/utils/timed-out.ts

```diff
@@ -110,15 +113,15 @@ export default (request: ClientRequest, delays: Delays, options: TimedOutOptions
 
 		/* istanbul ignore next: hard to test */
 		if (socket.connecting) {
-			if (delays.lookup !== undefined && !socketPath && !net.isIP(hostname || host)) {
+			if (delays.lookup !== undefined && !socketPath && !net.isIP(hostname || host || '')) {
 				const cancelTimeout = addTimeout(delays.lookup, timeoutHandler, 'lookup');
 				once(socket, 'lookup', cancelTimeout);
 			}
 
 			if (delays.connect !== undefined) {
-				const timeConnect = (): (() => void) => addTimeout(delays.connect, timeoutHandler, 'connect');
+				const timeConnect = (): (() => void) => addTimeout(delays.connect!, timeoutHandler, 'connect');
 
-				if (socketPath || net.isIP(hostname || host)) {
+				if (socketPath || net.isIP(hostname || host || '')) {
```

Comment:
I see that this net.isIP check is used at least twice, can't we abstract it to one central variable?